File: frontend-vue3-vite/src/api/room.ts
import http from './http'

export interface Room {
  id: number
  name: string
  description?: string
  isPublic: boolean
  ownerId: number
  ownerName: string
  memberCount: number
  inviteCode?: string
  createdAt: string
  messageCount?: number
}

export interface RoomMember {
  id: number
  roomId: number
  userId: number
  userName: string
  userAvatar?: string
  role: 'owner' | 'admin' | 'member'
  joinedAt: string
}

export interface RoomMessage {
  id: number
  content: string
  userId: number
  userName: string
  userAvatar?: string
  roomId: number
  createdAt: string
}

export interface RoomListResponse {
  rooms: Room[]
  total: number
}

export interface MemberListResponse {
  members: RoomMember[]
  total: number
}

export interface MessageListResponse {
  messages: RoomMessage[]
  total: number
  page: number
  limit: number
}

export interface CreateRoomData {
  name: string
  description?: string
  isPublic: boolean
}

export interface UpdateRoomData {
  name?: string
  description?: string
  isPublic?: boolean
}

export const roomApi = {
  getMyRooms() {
    return http.get<RoomListResponse>('/social/rooms/my')
  },

  getPublicRooms() {
    return http.get<RoomListResponse>('/social/rooms/public')
  },

  getRoom(roomId: number) {
    return http.get<Room>(`/social/rooms/${roomId}`)
  },

  createRoom(data: CreateRoomData) {
    return http.post<Room>('/social/rooms', data)
  },

  updateRoom(roomId: number, data: UpdateRoomData) {
    return http.put<Room>(`/social/rooms/${roomId}`, data)
  },

  deleteRoom(roomId: number) {
    return http.delete(`/social/rooms/${roomId}`)
  },

  joinRoom(roomId: number) {
    return http.post(`/social/rooms/${roomId}/join`)
  },

  joinRoomByCode(inviteCode: string) {
    return http.post('/social/rooms/join-by-code', { inviteCode })
  },

  leaveRoom(roomId: number) {
    return http.post(`/social/rooms/${roomId}/leave`)
  },

  getRoomMembers(roomId: number) {
    return http.get<MemberListResponse>(`/social/rooms/${roomId}/members`)
  },

  addRoomMember(roomId: number, userId: number) {
    return http.post(`/social/rooms/${roomId}/members`, { userId })
  },

  removeRoomMember(roomId: number, userId: number) {
    return http.delete(`/social/rooms/${roomId}/members/${userId}`)
  },

  getRoomMessages(roomId: number, params?: { page?: number; limit?: number }) {
    const { page = 1, limit = 50 } = params || {}
    return http.get<MessageListResponse>(`/social/rooms/${roomId}/messages?page=${page}&limit=${limit}`)
  },

  sendRoomMessage(roomId: number, content: string) {
    return http.post<RoomMessage>(`/social/rooms/${roomId}/messages`, { content })
  },

  deleteRoomMessage(roomId: number, messageId: number) {
    return http.delete(`/social/rooms/${roomId}/messages/${messageId}`)
  }
}

export default roomApi

File: backend-C#/Models/Room.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace IMS.Models;

[Table("rooms")]
public class Room
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Required]
    [Column("name")]
    [StringLength(100)]
    public string Name { get; set; } = string.Empty;

    [Column("description")]
    public string? Description { get; set; }

    [Required]
    [Column("owner_id")]
    public long OwnerId { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = true;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    [Column("deleted_at")]
    public DateTime? DeletedAt { get; set; }

    // Navigation properties
    [ForeignKey("OwnerId")]
    public User? Owner { get; set; }
    public ICollection<RoomMember>? Members { get; set; }
    public ICollection<RoomMessage>? Messages { get; set; }
}

[Table("room_members")]
public class RoomMember
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Required]
    [Column("room_id")]
    public long RoomId { get; set; }

    [Required]
    [Column("user_id")]
    public long UserId { get; set; }

    [Column("role")]
    [StringLength(20)]
    public string Role { get; set; } = "member";

    [Column("joined_at")]
    public DateTime JoinedAt { get; set; } = DateTime.UtcNow;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    [Column("deleted_at")]
    public DateTime? DeletedAt { get; set; }

    // Navigation properties
    [ForeignKey("RoomId")]
    public Room? Room { get; set; }

    [ForeignKey("UserId")]
    public User? User { get; set; }
}

[Table("room_messages")]
public class RoomMessage
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Required]
    [Column("room_id")]
    public long RoomId { get; set; }

    [Required]
    [Column("sender_id")]
    public long SenderId { get; set; }

    [Required]
    [Column("content")]
    public string Content { get; set; } = string.Empty;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    [Column("deleted_at")]
    public DateTime? DeletedAt { get; set; }

    // Navigation properties
    [ForeignKey("RoomId")]
    public Room? Room { get; set; }

    [ForeignKey("SenderId")]
    public User? Sender { get; set; }
}
